nengo/dists.py

```diff
@@ -311,6 +311,43 @@ def sample(self, n, d=None, rng=np.random):
         return self.options[i]
 
 
+class Samples(Distribution):
+    """A set of samples.
+
+    This class is a subclass of `.Distribution` so that it can be used in any
+    situation that calls for a  `.Distribution`. However, the call to `.sample`
+    must match the dimensions of the samples or a `.ValidationError`
+    will be raised.
+
+    Parameters
+    ----------
+    samples : (n, d) array_like
+        ``n`` and ``d`` must match what is eventually passed to `.sample`.
+    """
+
+    samples = NdarrayParam('samples', shape=('...',))
+
+    def __init__(self, samples):
+        super(Samples, self).__init__()
+        self.samples = samples
+
+    def __repr__(self):
+        return "Samples(samples=%r)" % self.samples
```

Comment:
As above, `samples` is always a numpy array here so we should be good. I considered the defensive tuple though, so I can slip it in there for kicks.